File: desktop/windows/Nexus.Desktop/Sidecar/DesktopPaths.cs
using System.IO;

namespace Nexus.Desktop.Sidecar;

internal static class DesktopPaths
{
    public static string RootDirectory =>
        Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".nexus");

    public static string DataDirectory => Path.Combine(RootDirectory, "data");

    public static string ApplicationDataDirectory => RootDirectory;

    public static string ConfigDirectory => Path.Combine(RootDirectory, "config");

    public static string WorkspaceDirectory => Path.Combine(RootDirectory, "workspace");

    public static string CacheDirectory => Path.Combine(RootDirectory, "cache");

    public static string LogsDirectory => Path.Combine(RootDirectory, "logs");
}

File: desktop/macos/Sources/NexusDesktop/Runtime/DesktopPaths.swift
import Foundation

enum DesktopPaths {
  static var rootDirectory: URL {
    URL(fileURLWithPath: NSHomeDirectory()).appendingPathComponent(".nexus", isDirectory: true)
  }

  static var dataDirectory: URL {
    rootDirectory.appendingPathComponent("data", isDirectory: true)
  }

  static var configDirectory: URL {
    rootDirectory.appendingPathComponent("config", isDirectory: true)
  }

  static var workspaceDirectory: URL {
    rootDirectory.appendingPathComponent("workspace", isDirectory: true)
  }

  static var cacheDirectory: URL {
    rootDirectory.appendingPathComponent("cache", isDirectory: true)
  }

  static var logsDirectory: URL {
    rootDirectory.appendingPathComponent("logs", isDirectory: true)
  }

  static var sidecarPIDFileURL: URL {
    rootDirectory.appendingPathComponent("NexusSidecar.pid.json")
  }

  static var connectorCredentialsFallbackKeyURL: URL {
    configDirectory.appendingPathComponent("connector-credentials.key")
  }

  static func createRuntimeDirectories() throws {
    for directory in [rootDirectory, dataDirectory, configDirectory, workspaceDirectory, cacheDirectory, logsDirectory] {
      try FileManager.default.createDirectory(at: directory, withIntermediateDirectories: true)
    }
  }
}
